tools/c7n_azure/c7n_azure/function_package.py

```diff
@@ -31,6 +28,25 @@
 from c7n_azure.dependency_manager import DependencyManager
 from c7n_azure.session import Session
 
+from c7n.mu import PythonPackageArchive
+from c7n.utils import local_session
+
+
+class AzurePythonPackageArchive(PythonPackageArchive):
+    def __init__(self, modules=(), cache_file=None):
+        super(AzurePythonPackageArchive, self).__init__(modules, cache_file)
+        self.package_time = time.gmtime()
+
+    def create_zinfo(self, fname):
+        """
+        In Dedicated App Service Plans - Functions are updated via KuduSync
+        The KuduSync uses the modified time and file size to determine if a file has changed
+        """
+        info = super(AzurePythonPackageArchive, self).create_zinfo(fname)
+
+        info.date_time = self.package_time[0:6]
```

Comment:
Feels a little odd to override the zinfo creation instead of adding a `use_file_timestamps=False` parameter on `PythonPackageArchive`.

Maybe it makes more sense if you just move the other two zinfo property sets into `create_zinfo`, e.g.

```
      # Ensure we apply the compression
        dest.compress_type = self.zip_compression
        # Mark host OS as Linux for all archives
        dest.create_system = 3
```

Seems weird to just have it set the one property we happen to need to override at the moment.

An argument could be made we should be using a real file last modified timestamp so kudusync works the way its intended, like it wouldn't re-copy all the dep files unnecessarily, but this is not a major concern.

Reply:
It should not re-copy all the dep files because we copy bytes from cached zip file, so timestamps for the dependencies are persisted.

Reply:
I'm ok with either approach. If you do override (which should be extend, not override imo), it makes sense to move this to the original create_zinfo() too (and call it from child).

```
        # Ensure we apply the compression
        dest.compress_type = self.zip_compression
        # Mark host OS as Linux for all archives
        dest.create_system = 3
```

Reply:
Yeah agreed lets just move those bits in so its consistent.

Reply:
Like @logachev said if we passed a cached zip, those files won't be updated. I will move those settings into the `create_zinfo()`. 

I prefer not adding this in an option into the the `PythonPackageArchive`, because it's an option that may have negative consequences for the other providers with their current implementation.

Reply:
Yeah that would default to no-op and just have a conditional in add-contents but I agree option 2 is good.   

re: cache zip - I was sort of thinking file timestamps might flow all the way from the wheels in pypi and not just the cache zip file on one persons machine, but not worried about this :)